framework/entity/src/main/java/org/apache/ofbiz/entity/util/EntityUtilProperties.java

```diff
@@ -123,6 +123,117 @@ public static String getPropertyValueFromDelegatorName(String resource, String n
         }
     }
 
+    public static <T> Object getPropertyValue(String resource, String name, Object defaultValue, Delegator delegator, Class<T> clazz) {
+        Map<String, String> propMap = getSystemPropertyValue(resource, name, delegator);
+        if ("Y".equals(propMap.get("isExistInDb"))) {
+            String s = propMap.get("value");
+            if (UtilValidate.isEmpty(s)) {
+                return defaultValue;
+            }
+            try {
+                if (clazz.componentType() == Integer.class) {
```

Comment:
Hello, since it is for trunk and java 17 is available, what do you think about using switch pattern matching (and Guarded Pattern) 
ref : https://openjdk.org/jeps/406
Thanks and regards, 
Gil

Reply:
Mmm, i got carried away discovering those new switch syntax :), simple switch are sufficient here, sorry.

Reply:
Hey gil, 

i closed this PR and opened another one because the commit description is wrong. The new PR: https://github.com/apache/ofbiz-framework/pull/635
Also i will adjust the existing code with a switch in an upcoming commit in the new PR.

Thanks and regards,
Tobias

Reply:
Thanks to you Tobias :)

Reply:
Yes pattern matching is not applicable here, hence my second comment, sorry for the misunderstanding.  If in those cases are fine, since we are operating on `Class` object test...
Thanks